pytest_arraydiff/plugin.py

```diff
@@ -310,8 +310,10 @@ def pytest_runtest_call(self, item):
         # Find test name to use as plot name
         filename = compare.kwargs.get('filename', None)
         if filename is None:
-            filename = item.name + '.' + extension
-            if not single_reference:
+            if single_reference:
+                filename = item.function + '.' + extension
```

Comment:
Whoops, it was accessing the name before https://github.com/astropy/pytest-arraydiff/commit/04e365cfbdc7992fbd2772107726706942aae374#diff-c5983910b8498a8d2d64db7dd4bf7a2cf6e7399a31a161fd2e5eed2f04484460L286
```suggestion
                filename = item.function.__name__ + '.' + extension
```

Reply:
Let's try!